eclair-node/src/main/scala/fr/acinq/eclair/api/directives/ExtraDirectives.scala

```diff
@@ -47,7 +47,7 @@ trait ExtraDirectives extends Directives {
   val outPointsFormParam: NameUnmarshallerReceptacle[List[OutPoint]] = "outpoints".as[List[OutPoint]](outPointListUnmarshaller)
   val paymentHashFormParam: NameUnmarshallerReceptacle[ByteVector32] = "paymentHash".as[ByteVector32](bytes32Unmarshaller)
   val amountMsatFormParam: NameReceptacle[MilliSatoshi] = "amountMsat".as[MilliSatoshi]
-  val invoiceFormParam: NameReceptacle[Bolt11Invoice] = "invoice".as[Bolt11Invoice]
+  val invoiceFormParam: NameUnmarshallerReceptacle[Invoice] = "invoice".as[Invoice](invoiceUnmarshaller)
```

Comment:
We should keep this for Bolt 11 only, Bolt 12 invoices should be obtained with the `invoice_request` flow and never accepted as a parameter?

Reply:
From the spec point of view, Bolt 12 invoices should *not* be exposed to end users, only the *offer* should be exposed. You shouldn't be able to explicitly pay a Bolt 12 invoice, APIs should only allow you to pay a Bolt 12 offer.

So I'm not sure why you'd want to be able to parse them, since they're meant to be internal objects that aren't exposed to end users. On top of that, they're just a TLV stream, so it won't look nice in JSON at all...